### PR TITLE
RWA-2273-remove available tasks only from search parameter

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-2273

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java [hunk 1]
```diff
@@ -224,35 +220,17 @@ private PermissionRequirements findPermissionRequirement(SearchTaskRequest searc
...
-        SearchParameterBoolean availableTasksOnly = boolKeyMap.get(AVAILABLE_TASKS_ONLY);
 
         RequestContext context = searchTaskRequest.getRequestContext();
 
         if (context == null) {
-            return availableTasksOnly != null && availableTasksOnly.getValues();
+            return false;
         } else {
             return context.equals(RequestContext.AVAILABLE_TASKS);
         }
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java:231]: you can simplify this like - return context != null && ontext.equals(RequestContext.AVAILABLE_TASKS);

**Review by rasikasri — APPROVED** [review]

**Review by rasikasri — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java [hunk 2]
```diff
@@ -1258,7 +1294,8 @@ void should_return_400_bad_request_when_available_tasks_only_is_set_with_empty_v
                     jsonPath("$.title").value("Bad Request"),
                     jsonPath("$.status").value(400),
                     jsonPath("$.detail").value("Invalid request field: search_parameters.[0]: "
-                                               + "Unexpected character ('}' (code 125)): expected a value")
+                                               + "Each search_parameter element must have "
+                                               + "'key', 'values' and 'operator' fields present and populated.")
                 ));
     }
```
Comment (PrasadKumarP) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java:1301]: It should be available_task

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java [hunk 3]
```diff
@@ -904,14 +902,16 @@ private static Stream<TaskQueryScenario> grantTypeWithChallengedAndExcludedScena
     }
 
     private static Stream<TaskQueryScenario> grantTypeWithAvailableTasksOnlyScenarioHappyPath() {
```
Comment (PrasadKumarP) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java:904]: change method name to AavailableTasks

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/wa/CftQueryServiceITTest.java [hunk 4]
```diff
@@ -736,14 +734,15 @@ private static Stream<TaskQueryScenario> grantTypeWithChallengedAndExcludedError
     }
 
     private static Stream<TaskQueryScenario> grantTypeWithAvailableTasksOnlyScenarioHappyPath() {
```
Comment (PrasadKumarP) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/wa/CftQueryServiceITTest.java:736]: change method name to AavailableTasks

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java [hunk 5]
```diff
@@ -188,14 +188,20 @@ private PermissionRequirements findPermissionRequirement(SearchRequest searchReq
             }
             return PermissionRequirementBuilder.builder().buildSingleType(READ);
         } else {
-            if (searchRequest.isAvailableTasksOnly()) {
+            if (isAvailableTasksOnly(searchRequest)) {
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java:191]: You can use the isAvailableTaskOnly method in the search request now.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java [hunk 6]
```diff
@@ -188,14 +188,20 @@ private PermissionRequirements findPermissionRequirement(SearchRequest searchReq
...
                 return PermissionRequirementBuilder.builder().buildSingleRequirementWithAnd(OWN, READ);
             } else {
                 return PermissionRequirementBuilder.builder().buildSingleType(READ);
             }
         }
     }
 
+    private boolean isAvailableTasksOnly(SearchRequest searchTaskRequest) {
```
Comment (rasikasri) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java:199]: We don't need this method now, as we have it in the search request already

**Review by rasikasri — APPROVED** [review]

**Pull request merged**